docs/SIL.rst

```diff
@@ -5707,6 +5707,49 @@ Extracts the original function or a transpose function from the given
 ``[original]`` or ``[transpose]``.
 
 
+differentiability_witness_function
+``````````````````````````````````
+
+::
+
+  sil-instruction ::= 'differentiability_witness_function'
+                      sil-differentiability-witness-function-kind
+                      '[' 'parameters' sil-differentiability-witness-indices ']'
+                      '[' 'results' sil-differentiability-witness-indices ']'
+                      generic-parameter-clause?
+                      sil-function-name ':' sil-type
+
+  sil-differentiability-witness-function-kind ::=
+      '[' sil-differentiability-witness-function-kind-name ']'
+  sil-differentiability-witness-function-kind-name ::=
+      'jvp' | 'vjp' | 'transpose'
+  sil-differentiability-witness-indices ::= [0-9]+ (' ' [0-9]+)*
+  generic-parameter-clause ::=
```

Comment:
Add a "sil-differentiability-witness-function-" prefix to all non-terminals.
```suggestion
  sil-differentiability-witness-function-generic-parameter-clause ::=
```

Reply:
Since `generic-parameter-clause` is a well-defined in [libSyntax](https://github.com/apple/swift/blob/92b0f221113ba202f113716b0b245246b8da2cd9/utils/gyb_syntax_support/GenericNodes.py#L70), I chose to omit it here.

There are no other references to `generic-parameter-clause` in SIL.rst, only `generic-parameter-list` (which is one subcomponent).